scenario/runtime.py

```diff
@@ -58,6 +54,192 @@ class DirtyVirtualCharmRootError(ScenarioRuntimeError):
     """Error raised when the runtime can't initialize the vroot without overwriting existing metadata files."""
 
 
+class InconsistentScenarioError(ScenarioRuntimeError):
+    """Error raised when the combination of state and event is inconsistent."""
+
+
+class ConsistencyChecker:
```

Comment:
When would we not want to do these checks?
Could/should the checks be move into where they are relevant, instead of being centralized here?

Reply:
we want to skip the checks if we're using scenario to test the framework or charm libs, for example if we want to check how they handle edge cases or juju bugs.

For example check `test_rubbish_event_raises`

Reply:
moved where they are relevant, as in, the respective State constructor? 
Some of these checks used to be more spread out, validating the inputs to State and so on, but I found it confusing to keep in mind what was already validated and what was not validated.
Besides, some of the information we need is only available later (the charm spec aka the compound charm metadata), so we need to delay the check until we have all the data we need.

Reply:
Some basic input validation is done at State construction time, and if you pass invalid metadata that should also be surfaced earlier/elsewhere. This is really about cross-domain consistency. An overarching decision of whether this scenario 'makes sense' from a juju perspective, or whether the situation you're testing is nonsense as in: "it would practically never occur in a real-life charm".

Reply:
E.g. you try and fire a `foo-ready-event` but you don't even have a `foo` container in metadata.yaml.